server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java

```diff
@@ -194,28 +215,19 @@ private void createMetadataFile(VolumeManager volmanager, String fileName,
         .forFile(file, fs, fs.getConf(), cs).withTableConfiguration(conf).build();
     tabletWriter.startDefaultLocalityGroup();
 
-    for (Map.Entry<Key,Value> entry : sorted.entrySet()) {
-      tabletWriter.append(entry.getKey(), entry.getValue());
+    for (InitialTablet initialTablet : initialTablets) {
+      // sort file contents in memory, then play back to the file
+      for (Map.Entry<Key,Value> entry : initialTablet.createEntries().entrySet()) {
+        tabletWriter.append(entry.getKey(), entry.getValue());
+      }
```

Comment:
Still need to sort the key values prior to append them to file.  The file expects data appended to be in sorted order and will throw an exception if its not.  The order of the InitialTablets list may or may not  produce sorted data.  It probably is producing sorted data at the moment if init is running w/o issue.

```suggestion
   TreeMap<Key,Value> sorted = new TreeMap<>();

    for (InitialTablet initialTablet : initialTablets) {
      // sort file contents in memory, then play back to the file
      sorted.putAll( initialTablet.createEntries());
    }
    
    for (Map.Entry<Key,Value> entry : sorted.entrySet()) {
      tabletWriter.append(entry.getKey(), entry.getValue());
    }
```

Reply:
I used a treeMap inside of `InitalTablet.createEntries()` so it should be sorted for each individual tablet.
However, this section does use combined initalTablets output so there is no guarantee that it is sorted.

Added the sort logic back in for this.